src/main/java/org/example/chess/piece/Horse.java

```diff
@@ -0,0 +1,64 @@
+package org.example.chess.piece;
+
+import org.example.chess.Field;
+import org.example.chess.piece.util.ChessPieceFieldValidator;
+
+import java.util.HashSet;
+import java.util.Set;
+
+final class Horse extends ChessPiece{
+    Horse(Color color, Field currentPosition) {
+        super(color, currentPosition);
+    }
+
+    @Override
+    Set<Field> possibleMoves() {
+        Set<Field> fieldsOfPossibleMoves = new HashSet<>();
+
+        checkMoves(fieldsOfPossibleMoves, "up");
+        checkMoves(fieldsOfPossibleMoves, "down");
+        checkMoves(fieldsOfPossibleMoves, "right");
+        checkMoves(fieldsOfPossibleMoves, "left");
+
+        return fieldsOfPossibleMoves;
+    }
+
+
+    private static Set<Field> checkMoves(Set<Field> fields, String command) {
```

Comment:
It seems to me that as a rule of thumb we shouldn't pass an argument to a method that is later used to branch logic in the said method I think it falls under the principle that a function should have only only do one thing